### PR TITLE
Port couchbase and mongo db activites

### PR DESCRIPTION
**Pull request by skothari-tibco**

### CODE REVIEW

File: activity/couchbase/metadata.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 type Settings struct {
 	Username       string `md:"username"`
 	Password       string `md:"password"`
-	BucketName     string `md:"bucket,required"`
+	BucketName     string `md:"bucketname,required"`
```
Comment (fm-tibco) [inline activity/couchbase/metadata.go]: switch to camel case, bucketName and bucketPassword

File: activity/mongodb/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "id": "flogo-mongodb",
+  "name": "MongoDb",
+  "description": "MongoDb Activity",
+  "activity": {
+    "ref": "github.com/project-flogo/datastore-contrib/activity/mongodb",
+    "settings": {
+      "uri" : "localhost:27017"
+    },
+    "input" : {
+        "dbname" : "test",
```
Comment (fm-tibco) [inline activity/mongodb/README.md]: shouldn't this be dbName.. the others probably keyName, keyValue

File: activity/couchbase/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```bash
+flogo install github.com/project-flogo/datastore-contrib/activity/couchbase
+```
+
+## Configuration
+
+### Settings:
+| Name              | Type   | Description
+| :---              | :---   | :---
+| Username          | string |
```
Comment (fm-tibco) [inline activity/couchbase/README.md]: This should use the json keys

File: activity/couchbase/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      "server" : "http://localhost:8091",
+      "username": "Administator",
+      "password": "password",
+      "bucketname" : "sample",
+      "bucketpassword" : "",
+    },
+    "input" : {
+        "key" : "test",
+        "data" : "example",
+        "method" : "Insert",
```
Comment (fm-tibco) [inline activity/couchbase/README.md:58]: I would move method to settings, you don't typically change this dynamically

**Review by fm-tibco — APPROVED** [review]

File: activity/mongodb/README.md (hunk 2 above)
Comment (fm-tibco) [reply activity/mongodb/README.md]: Some more of these should probably move to Settings, they are the values that are typically fixed.  I would think probably DbName, Collection, Method.  Also these tables should use the json keys.

**Review by fm-tibco — CHANGES_REQUESTED** [review]

File: activity/mongodb/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+flogo install github.com/project-flogo/datastore-contrib/activity/mongodb
+```
+
+## Configuration
+
+### Settings:
+| Name     | Type   | Description
+|:---      | :---   | :---    
+| uri      | string | The MongoDB connection URI
+| method   | string | The method type (DELETE, INSERT, UPDATE or REPLACE)
```
Comment (fm-tibco) [inline activity/mongodb/README.md:17]: Can you update the settings

File: activity/couchbase/descriptor.json [hunk 6]
```diff
@@ -5,6 +5,30 @@
     "title": "Couchbase connector",
     "description": "Manage Couchbase documents in a specified bucket",
     "homepage": "https://github.com/project-flogo/contrib/tree/master/activity/couchbase",
+    "settings":[
```
Comment (fm-tibco) [inline activity/couchbase/descriptor.json:8]: Can you add descriptions to these.  Use the description from the README.md

File: activity/couchbase/README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+flogo install github.com/project-flogo/datastore-contrib/activity/couchbase
+```
+
+## Configuration
+
+### Settings:
+| Name              | Type   | Description
+| :---              | :---   | :---
+| Username          | string |     
+| Password          | string |
```
Comment (fm-tibco) [inline activity/couchbase/README.md]: Can you add descriptions to these

File: activity/mongodb/descriptor.json [hunk 8]
```diff
@@ -5,34 +5,36 @@
     "title": "MongoDB",
     "description": "Manage MongoDB documents in a specified collection",
     "homepage": "",
-    "input": [
+    "settings":[
```
Comment (fm-tibco) [inline activity/mongodb/descriptor.json:8]: Can you add descriptions to these.  Look at examples under contrib

**Review by fm-tibco — CHANGES_REQUESTED** [review]

File: activity/couchbase/README.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+```bash
+flogo install github.com/project-flogo/datastore-contrib/activity/couchbase
+```
+
+## Configuration
+
+### Settings:
+| Name              | Type   | Description
+| :---              | :---   | :---
+| Username          | string | Cluster username
```
Comment (fm-tibco) [inline activity/couchbase/README.md]: This should juse the json keys

**Review by fm-tibco — CHANGES_REQUESTED** [review]

File: activity/couchbase/README.md (hunk 9 above)
Comment (skothari-tibco) [reply activity/couchbase/README.md]: Ohh. Got it

**Review by fm-tibco — APPROVED** [review]

**Pull request merged**